Sources/SwiftInspectorVisitors/StructVisitor.swift

```diff
@@ -0,0 +1,121 @@
+      let qualifiedParentTypeName: String
+      if let parentTypeName = parentTypeName {
+        qualifiedParentTypeName = "\(parentTypeName).\(structInfo.name)"
+      } else {
+        qualifiedParentTypeName = structInfo.name
+      }
+
+      let innerStructVisitor = StructVisitor(parentTypeName: qualifiedParentTypeName)
+      innerStructVisitor.walk(node)
+
+      self.innerStructs += innerStructVisitor.structs
+      // We've already gotten information from the children from our inner struct visitor.
+      return .skipChildren
+
+    } else {
+      // Recursive case. This is the first struct declaration we've come across.
+      // We need to get its information and then visit children to see if there is more information we need.
+      let name = node.identifier.text
+      let typeInheritanceVisitor = TypeInheritanceVisitor()
+      typeInheritanceVisitor.walk(node)
+
+      structInfo = StructInfo(
+        name: name,
+        inheritsFromTypes: typeInheritanceVisitor.inheritsFromTypes,
+        parentTypeName: parentTypeName)
+      return .visitChildren
+    }
+  }
+
+  public override func visitPost(_ node: StructDeclSyntax) {
+    hasFinishedParsingStruct = node.identifier.text == structInfo?.name
+  }
+
+  public override func visit(_ node: ClassDeclSyntax) -> SyntaxVisitorContinueKind {
+    if let _ = structInfo {
+      // We've previously found a struct declaration, so this must be an inner class.
+      // TODO: Utilize class visitor to find inner class information, utilizing parent information from structInfo
+    } else {
+      // We've encountered a class declaration before encountering a struct declaration. Something is wrong.
+      assertionFailure("Encountered a top-level class. This is a usage error: a single StructVisitor instance should start walking only over a node of type `StructDeclSyntax`")
+    }
+    return .skipChildren
+  }
+
+  public override func visit(_ node: EnumDeclSyntax) -> SyntaxVisitorContinueKind {
+    if let _ = structInfo {
+      // We've previously found a struct declaration, so this must be an inner enum.
+      // TODO: Utilize enum visitor to find inner enum information, utilizing parent information from structInfo
+    } else {
+      // We've encountered a class declaration before encountering a struct declaration. Something is wrong.
+      assertionFailure("Encountered a top-level class. This is a usage error: a single StructVisitor instance should start walking only over a node of type `StructDeclSyntax`")
+    }
+    return .skipChildren
+
+  }
+
+  // MARK: Private
+
+  private let parentTypeName: String?
+  private var hasFinishedParsingStruct = false
+  private var structInfo: StructInfo?
+  private var innerStructs = [StructInfo]()
+}
+
+public struct StructInfo: Codable, Equatable {
+  public let name: String
+  public let inheritsFromTypes: [String]
+  public let parentTypeName: String?
+  // TODO: also find and expose properties on a struct
```

Comment:
When we eventually start parsing properties, we'll be able to visit variable declarations and add them to our `structInfo` property. We won't accidentally scoop up variable declarations from inner structs because every instance of `StructVisitor` stops parsing children when it hits another struct.